src/main/java/uk/gov/hmcts/reform/sandl/snlrules/exception/DateComparisonException.java

```diff
@@ -0,0 +1,7 @@
+package uk.gov.hmcts.reform.sandl.snlrules.exception;
+
+public class DateComparisonException extends Exception {
```

Comment:
checked exception.... think twice and let me know if you still think this is a good idea.
recommended reading https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-exceptions.html